packages/apps/job-launcher/server/src/modules/job/job.service.ts

```diff
@@ -0,0 +1,373 @@
+      submissionsRequired: annotationsPerImage,
+      labels,
+      requesterDescription,
+      requesterAccuracyTarget,
+      fundAmount: totalAmount.toString(),
+      mode: JobMode.BATCH,
+      requestType: JobRequestType.IMAGE_LABEL_BINARY,
+    };
+
+    const { manifestUrl, manifestHash } = await this.saveManifest(
+      manifestData,
+      this.bucket,
+    );
+
+    const jobEntity = await this.jobRepository.create({
+      chainId,
+      userId,
+      manifestUrl,
+      manifestHash,
+      status: JobStatus.PENDING,
+      waitUntil: new Date(),
+    });
+
+    if (!jobEntity) {
+      this.logger.log(ErrorJob.NotCreated, JobService.name);
+      throw new NotFoundException(ErrorJob.NotCreated);
+    }
+
+    await this.paymentService.savePayment(
+      userId,
+      PaymentSource.BALANCE,
+      PaymentType.WITHDRAWAL,
+      BigNumber.from(fundAmount),
+    );
+
+    jobEntity.status = JobStatus.PAID;
+    await jobEntity.save();
+
+    return jobEntity.id;
+  }
+
+  public async launchJob(jobEntity: JobEntity): Promise<JobEntity> {
+    try {
+      const provider = new providers.JsonRpcProvider(
+        Object.values(networkMap).find(
+          (item) => item.network.chainId === jobEntity.chainId,
+        )?.rpcUrl,
+      );
+
+      const signer = new Wallet(
+        this.configService.get<string>(
+          'WEB3_JOB_LAUNCHER_PRIVATE_KEY',
+          'web3_private_key',
+        ),
+        provider,
+      );
+      
+      const clientParams = await InitClient.getParams(signer);
+
+      const escrowClient = new EscrowClient(clientParams);
+
+      const escrowConfig = {
+        recordingOracle: this.configService.get<string>(
+          'RECORDING_ORACLE_ADDRESS',
+          '',
+        ),
+        reputationOracle: this.configService.get<string>(
+          'REPUTATION_ORACLE_ADDRESS',
+          '',
+        ),
+        recordingOracleFee: BigNumber.from(this.configService.get<number>(
+          'RECORDING_ORACLE_FEE',
+          0,
+        )),
+        reputationOracleFee: BigNumber.from(this.configService.get<number>(
+          'REPUTATION_ORACLE_FEE',
+          0,
+        )),
+        manifestUrl: jobEntity.manifestUrl,
+        manifestHash: jobEntity.manifestHash,
+      };
+
+      const escrowAddress = await escrowClient.createAndSetupEscrow(
+        clientParams.network.hmtAddress,
+        [],
+        escrowConfig,
+      );
+
+      if (!escrowAddress) {
+        this.logger.log(ErrorEscrow.NotCreated, JobService.name);
+        throw new NotFoundException(ErrorEscrow.NotCreated);
+      }
+
+      jobEntity.escrowAddress = escrowAddress;
+      jobEntity.status = JobStatus.LAUNCHED;
+      await jobEntity.save();
+
+      const manifest = await this.getManifest(jobEntity.manifestUrl);
+
+      if (manifest.requestType === JobRequestType.IMAGE_LABEL_BINARY) {
+        this.sendWebhook(
+          this.configService.get<string>(
+            'EXCHANGE_ORACLE_WEBHOOK_URL',
+            '',
```

Comment:
if webhook fails it throws an error but the escrow was actually launched

Reply:
We need to add additional methods for process some of this exceptions